### DISCUSSION TITLE
How to output subpage content in a layout file?

### DISCUSSION THREAD

**Discussion by pftnhr** [datenstrom/community#848] [Ask a question] [answered]
I would like to output the content of several sub folders with something like `<?php echo $this->yellow->page->getPage("producta")->getContent() ?>` on one page, e.g. `shop.html`. I know that `getPage()` only works with shared pages, but I hope that I don't have to put different products in one level in the `Shared` folder to keep some structure.

Is there a decisive hint as to how I can solve this most elegantly?

Example folder structure:
```
|----- content               
|    |----- 1-home
|    |----- 5-shop
|         |----- 1-producta
|         |----- 2-product
|    |----- 9-about           
|    |----- shared            
|----- media                    
|----- system
|    |----- extensions
|    |----- layouts
|         |----- ...
|         |----- shop.html
|         |----- ...
|    |----- themes
|    |----- trash
```

**Comment by markseuffert** [comment, marked as answer]
The class [YellowContent](https://datenstrom.se/yellow/help/api-for-developers#yellow-content) gives access to content files. This example outputs one specific page:

```
<h1><?php echo $this->yellow->content->find("/shop/producta/")->getHtml("title") ?></h1>
<?php echo $this->yellow->content->find("/shop/producta/")->getContent() ?>
```

Subpages are also called children. This example snippet outputs multiple pages in a loop:

```
<?php $pages = $this->yellow->content->find("/shop/")->getChildren() ?>
<?php foreach ($pages as $page): ?>
<h1><?php echo $page->getHtml("title") ?></h1>
<?php echo $page->getContent() ?>
<?php endforeach ?>
```

You could also use the current page as starting position. Another example, the loop stays the same:

```
<?php $pages = $this->yellow->page->getChildren() ?>
<?php foreach ($pages as $page): ?>
<h1><?php echo $page->getHtml("titleContent") ?></h1>
<?php echo $page->getContent() ?>
<?php endforeach ?>
```

There are many possibilities to combine the API, you figure something out that works for you.

**Reply by pftnhr** [reply]
> ```
> <h1><?php echo $this->yellow->content->find("/shop/producta/")->getHtml("title") ?></h1>
> <?php echo $this->yellow->content->find("/shop/producta/")->getContent() ?>
> ```

This one works best for my needs. Thanks a lot Mark!
